records/generator.go

```diff
@@ -120,54 +128,56 @@ func (rg *RecordGenerator) loadFromMaster(ip string, port string) (sj state.Stat
 	req, err := http.NewRequest("GET", u.String(), nil)
 	if err != nil {
 		logging.Error.Println(err)
+		return nil, err
 	}
 
 	req.Header.Set("Content-Type", "application/json")
+	logging.VeryVerbose.Println("Timeout: " + rg.httpClient.Timeout.String())
 
-	client := &http.Client{}
-	resp, err := client.Do(req)
+	resp, err := rg.httpClient.Do(req)
 	if err != nil {
 		logging.Error.Println(err)
+		return nil, err
 	}
 
 	body, err := ioutil.ReadAll(resp.Body)
```

Comment:
should really have a `defer resp.Body.close()` here, before the `ReadAll` (and get rid of the one on line 148)